apps/common-app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/TestRunner.ts

```diff
@@ -36,15 +37,42 @@ function notifyJS(name: string) {
   notificationRegistry[name] = true;
 }
 
+export class RenderLock {
+  private _renderLock: LockObject = { lock: false };
+
+  public lockRender() {
+    this._renderLock = { lock: true };
+  }
+
+  public unlockRender() {
+    this._renderLock = { lock: false };
+  }
+
+  public waitForUnlock(maxWaitTime?: number) {
```

Comment:
is it the same implementation as `waitForThreadUnlock` from `SyncUIRunner`? Can we avoid code duplication? For example you can use inheritance